src/compiler/utilities.ts

```diff
@@ -11360,18 +11360,23 @@ export function createNameResolver({
                         location = root.parent;
                     }
                     break;
-                case SyntaxKind.Parameter:
+                case SyntaxKind.Parameter: {
+                    const param = location as ParameterDeclaration;
+                    if (lastLocation === param.name) {
+                        lastSelfReferenceLocation = param;

```

Comment:
I'd almost prefer a new parameter and entry in `isSelfReferenceLocation` like

```ts
    function isSelfReferenceLocation(node: Node, lastLocation: Node): node is SelfReferenceLocation {
        switch (node.kind) {
            case SyntaxKind.Parameter: return lastLocation === (node as ParameterDeclaration).name
            // ...
```

Reply:
I just pushed out the suggested change